source/cursorManager.py

```diff
@@ -365,14 +368,23 @@ def script_selectAll(self,gesture):
 		self._selectionMovementScriptHelper(toPosition=textInfos.POSITION_ALL)
 
 	def script_copyToClipboard(self,gesture):
-		info=self.makeTextInfo(textInfos.POSITION_SELECTION)
-		if info.isCollapsed:
-			# Translators: Reported when there is no text selected (for copying).
-			ui.message(_("No selection"))
+		info = self.makeTextInfo(textInfos.POSITION_SELECTION)
+		if not info.isCollapsed:
+			if info.copyToClipboard():
+				# Translators: Message presented when text has been copied to clipboard.
+				ui.message(_("Copied to clipboard"))
 			return
-		if info.copyToClipboard():
-			# Translators: Message presented when text has been copied to clipboard.
-			ui.message(_("Copied to clipboard"))
+		if isinstance(self, TreeInterceptor):
+			info = self.rootNVDAObject.makeTextInfo(textInfos.POSITION_SELECTION)
+			if not info.isCollapsed:
+				# There is indeed a selection, but L{info} strangely enough
+				# does not contains it (tested with Firefox and Chrome).
+				KeyboardInputGesture.fromName("control+c").send()
+				# Translators: Message presented when text has been copied to clipboard.
+				ui.message(_("Copied to clipboard"))

```

Comment:
Could copying text to the clipboard, even when there is a selection, be blocked by javascript? This could also be the case for right click, for example. In this case, this message would be announced while in fact, nothing is copied.

Reply:
Unless the CompboundTextInfo thing allows to retrieve the selected text, I agree we hardly can tell if the operation was successful. Even examining the clipboard could lead to false positives.
Would you please have a suggestion for an alternative announce?

Again, I'm not sure why we decided to have `control+c` bound in the first place, while everywhere else it is `NVDA+F10*2` that does the trick…

Reply:
I think it just has been added as such in a very early stage.

Reply:
> I think it just has been added as such in a very early stage.

I would really be in favor of removing this script altogether, unless there are corner cases I haven't realized, or unless we can come up with a fully functional implementation.

Reply:
Note that reviewing OCR results also involves a cursor manager, it makes sense to have this script there. It is also helpful for people to copy just small portions of text from browse mode, i.e. a news article without all the formatting. Still, ctrl+c might not be the most appropriate gesture.

Reply:
> Note that reviewing OCR results also involves a cursor manager, it makes sense to have this script there.

By the way, I would really prefer some kind of displayed browsable message for the OCR results…

> It is also helpful for people to copy just small portions of text from browse mode, i.e. a news article without all the formatting. Still, ctrl+c might not be the most appropriate gesture.

Dumb me: I did not realize `shift+[control+]arrow` was functional in browse mode…